src/components/Filters/SchemaSieve.ts

```diff
@@ -142,3 +149,136 @@ export const getOperators = (schema: JSONSchema6, field: string) => {
 		label: value.getLabel(subSchema),
 	}));
 };
+
+// The inner recursive function used by `flattenSchema`. Mutates the
+// `accumulator` argument.
+const flattenAccumulator = (
+	schema: JSONSchema6,
+	delimiter: string,
+	parentKey: string = '',
+	accumulator?: JSONSchema6,
+) => {
+	if (!accumulator) {
+		accumulator = cloneDeep(schema);
+
+		if (schema.properties) {
+			accumulator.properties = {};
+		}
+	}
+
+	if (schema.anyOf) {
+		accumulator.anyOf = schema.anyOf.map(subSchema =>
+			flattenAccumulator(subSchema, delimiter, parentKey),
+		);
+	}
+
+	forEach(schema.properties, (value: JSONSchema6, key) => {
```

Comment:
How about using `reduce` instead of `forEach` and remove all `!` from the lambda?
PS: `reduce` is actually semantically more appropriate.